### PR TITLE
Better for loops

### PR DESCRIPTION
**Pull request by mvieth**
Range for loops with const auto reference

### CODE REVIEW

File: surface/include/pcl/surface/impl/gp3.hpp [hunk 1]
```diff
@@ -116,11 +116,11 @@ pcl::GreedyProjectionTriangulation<PointInT>::reconstructPolygons (std::vector<p
   if (!input_->is_dense)
   {
     // Skip invalid points from the indices list
-    for (std::vector<int>::const_iterator it = indices_->begin (); it != indices_->end (); ++it)
-      if (!std::isfinite ((*input_)[*it].x) ||
-          !std::isfinite ((*input_)[*it].y) ||
-          !std::isfinite ((*input_)[*it].z))
-        state_[*it] = NONE;
+    for (const auto& it : (*indices_))
```
Comment (larshg) [inline surface/include/pcl/surface/impl/gp3.hpp:119]:
```suggestion
    for (const auto& i : (*indices_))
```

File: surface/include/pcl/surface/impl/gp3.hpp [hunk 2]
```diff
@@ -878,7 +878,7 @@ pcl::GreedyProjectionTriangulation<PointInT>::reconstructPolygons (std::vector<p
           else
             angle_so_far = 0;
         }
-        for (const int &it : to_erase)
+        for (const auto &it : to_erase)
```
Comment (larshg) [inline surface/include/pcl/surface/impl/gp3.hpp:881]:
```suggestion
        for (const auto &i : to_erase)
```

File: keypoints/include/pcl/keypoints/impl/harris_3d.hpp [hunk 3]
```diff
@@ -291,9 +291,9 @@ pcl::HarrisKeypoint3D<PointInT, PointOutT, NormalT>::detectKeypoints (PointCloud
       std::vector<float> nn_dists;
       tree_->radiusSearch (idx, search_radius_, nn_indices, nn_dists);
       bool is_maxima = true;
-      for (std::vector<int>::const_iterator iIt = nn_indices.begin(); iIt != nn_indices.end(); ++iIt)
+      for (const auto& iIt : nn_indices)
```
Comment (kunaltyagi) [inline keypoints/include/pcl/keypoints/impl/harris_3d.hpp:294]: index, not iterator

File: keypoints/include/pcl/keypoints/impl/harris_3d.hpp [hunk 4]
```diff
@@ -528,14 +528,14 @@ pcl::HarrisKeypoint3D<PointInT, PointOutT, NormalT>::refineCorners (PointCloudOu
       std::vector<int> nn_indices;
       std::vector<float> nn_dists;
       tree_->radiusSearch (corner, search_radius_, nn_indices, nn_dists);
-      for (std::vector<int>::const_iterator iIt = nn_indices.begin(); iIt != nn_indices.end(); ++iIt)
+      for (const auto& iIt : nn_indices)
```
Comment (kunaltyagi) [inline keypoints/include/pcl/keypoints/impl/harris_3d.hpp:531]: same here

File: keypoints/include/pcl/keypoints/impl/harris_6d.hpp [hunk 5]
```diff
@@ -238,9 +238,9 @@ pcl::HarrisKeypoint6D<PointInT, PointOutT, NormalT>::detectKeypoints (PointCloud
       std::vector<float> nn_dists;
       tree_->radiusSearch (idx, search_radius_, nn_indices, nn_dists);
       bool is_maxima = true;
-      for (std::vector<int>::const_iterator iIt = nn_indices.begin(); iIt != nn_indices.end(); ++iIt)
+      for (const auto& iIt : nn_indices)
```
Comment (kunaltyagi) [inline keypoints/include/pcl/keypoints/impl/harris_6d.hpp:241]: and in a lot of places 😄

**Comment by mvieth** [comment]
I think I fixed all names. Also I can squash later when you are ok with the changes.

**Review by kunaltyagi — APPROVED** [review]

**Review by kunaltyagi — APPROVED** [review]

File: surface/include/pcl/surface/impl/gp3.hpp [hunk 6]
```diff
@@ -878,10 +878,10 @@ pcl::GreedyProjectionTriangulation<PointInT>::reconstructPolygons (std::vector<p
           else
             angle_so_far = 0;
         }
-        for (const int &it : to_erase)
+        for (const auto &idx : to_erase)
         {
           for (std::vector<int>::iterator iter = angleIdx.begin(); iter != angleIdx.end(); ++iter)
```
Comment (larshg) [inline surface/include/pcl/surface/impl/gp3.hpp:883]: Range loop?

Comment (mvieth) [reply surface/include/pcl/surface/impl/gp3.hpp:883]: Not possible, because `iter` is needed for `erase` in line 884.

**Review by larshg — APPROVED** [review]

**Comment by kunaltyagi** [comment]
Squash and merge?

**Pull request merged**